src/librustc/hir/check_attr.rs

```diff
@@ -114,6 +118,31 @@ impl<'a, 'tcx> CheckAttrVisitor<'a, 'tcx> {
         }
     }
 
+    /// Check if the `#[marker]` attribute on an `item` is valid.
+    fn check_marker(&self, attr: &hir::Attribute, item: &hir::Item, target: Target) {
+        match target {
+            Target::Trait => { /* Valid */ },
+            _ => {
+                struct_span_err!(self.tcx.sess,
+                                 attr.span,
+                                 E0713,
+                                 "attribute can only be applied to a trait")
+                    .span_label(item.span, "not a trait")
+                    .emit();
+                return;
+            }
+        }
+
+        if attr.meta_item_list().is_some() || attr.value_str().is_some() {
```

Comment:
Is this enough? I know this is most likely copied from somewhere, but it seems the wrong way around (banning a few known attribute forms, instead of checking that it's a specific attribute form). cc @petrochenkov

Reply:
Yeah, `is_word()` should be more appropriate.

Reply:
Switched to `is_word`; thanks!